### ISSUE TITLE
The error info is incorrect when creating one table with the name 'Tables'

### ISSUE THREAD

**Issue by v-xianya** [microsoft/AzureStorageExplorer#5200]
**Storage Explorer Version**: 1.22.0-dev
**Build Number**: 20211121.3
**Branch**: main
**Platform/OS**: Windows 10/Linux Ubuntu 20.04/MacOS Monterey 12.0.1
**Architecture**: ia32/x64
**How Found**: From running test cases
**Regression From**: Previous release (1.21.3)

## Steps to Reproduce ##
1. Expand one storage account -> Right click 'Tables' node.
2. Create a table with the name 'Tables'.
3. Observe the error -> Check whether the error info is correct.

## Expected Experience ##
The error info is correct.
![image](https://user-images.githubusercontent.com/41351993/142817697-245e511f-8cd7-444b-bd76-0b585e1e301a.png)

## Actual Experience ##
The error info is incorrect.
![image](https://user-images.githubusercontent.com/41351993/142818201-fd2bf165-a086-4bdc-8dee-d34f5c673de3.png)

**Comment by craxal** [comment]
Unfortunately, not an easy fix. See, normally we'd expect every SDK function to throw an error that we can catch if the server response with an error, but `TableClient.createTable` doesn't throw an error for some reason. To work around this, I added a `tableExists` function that let's us return a meaningful error message.

The message is more cryptic for reserved table names, like "Tables". But apparently such tables do exist in the service, so the message is not technically wrong.

Since we're still returning reasonably correct error message, and we will need to wait on a fix in the SDK, I think it's safe to postpone fixing for a later release.

**Comment by craxal** [comment]
See Azure/azure-sdk-for-js#18865.

**Comment by craxal** [comment]
@v-xianya We updated the client library some time ago, and the link error is now closed. Can you verify whether this reproduces?

**Comment by v-xianya** [comment]
@craxal This issue still reproduces on the build main/20220206.3.
![image](https://user-images.githubusercontent.com/41351993/152713435-a8bd2992-7eab-4d12-922d-1d73f35a97ef.png)

**Comment by v-xianya** [comment]
@craxal 
Verified this issue on the build main/20220208.1. The current behaviors as below, is this expected?
1. A table named 'Tables' displays on the tree view. (Will disappear after manually refreshing)
![image](https://user-images.githubusercontent.com/41351993/152902613-c51ef6dd-2d45-4681-94f8-2823b4a5ae36.png)

2. The 'Tables' tab auto opens and the editor keeps loading.
![image](https://user-images.githubusercontent.com/41351993/152902626-259add8f-3ef5-46a3-b6b7-721beb3cda83.png)
3. There is a failed activity log. 

![image](https://user-images.githubusercontent.com/41351993/152902695-6846dc68-f94c-424c-800f-b77244e5904d.png)

**Comment by craxal** [comment]
@v-xianya The newly added node definitely needs to be fixed, but the error message comes straight from the client library and/or service. I think the error still communicates the essence of the problem and doesn't need to be changed.

**Comment by v-xianya** [comment]
Verified this issue on the build main/20220212.2. Fixed.
